core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,396 @@
+                    return PartitionSpecParser.toJson(spec);
+                  });
+
+      // iterate through delete manifests
+      CloseableIterable<ManifestFile> deleteManifests =
+          CloseableIterable.withNoopClose(snapshot().deleteManifests(tableOps().io()));
+      CloseableIterable<CloseableIterable<ScanTask>> results =
+          CloseableIterable.transform(
+              deleteManifests,
+              manifest -> {
+                // Filter partitions
+                CloseableIterable<ManifestEntry<DeleteFile>> deleteFileEntries =
+                    ManifestFiles.readDeleteManifest(manifest, tableOps().io(), transformedSpecs)
+                        .caseSensitive(isCaseSensitive())
+                        .select(scanColumns())
+                        .filterRows(rowFilter)
+                        .liveEntries();
+
+                // Filter delete file type
+                CloseableIterable<ManifestEntry<DeleteFile>> positionDeleteEntries =
+                    CloseableIterable.filter(
+                        deleteFileEntries,
+                        entry -> entry.file().content().equals(FileContent.POSITION_DELETES));
+
+                Types.StructType partitionType = Partitioning.partitionType(table());
+
+                return CloseableIterable.transform(
+                    positionDeleteEntries,
+                    entry -> {
+                      int specId = entry.file().specId();
+                      return new PositionDeleteScanTaskImpl(
+                          entry.file().copyWithoutStats(),
+                          schemaString,
+                          specStringCache.get(specId),
+                          residualCache.get(specId),
+                          partitionType);
+                    });
+              });
+
+      return new ParallelIterable<>(results, planExecutor());
+    }
+  }
+
+  /** Scan task for position delete files */
+  public interface PositionDeleteScanTask extends ContentScanTask<DeleteFile> {
+
+    /**
+     * Utility method to get constant values of rows to be scanned by this task. The following
+     * columns are constants of each PositionDeletesFileScanTask: * spec_id * partition *
+     * delete_file_path
+     *
+     * @return a map of column id to constant values returned by this task type
+     */
+    Map<Integer, ?> constantsMap();
+
+    /**
+     * Utility method to get constant values of rows to be scanned by this task. The following
+     * columns are constants of each PositionDeletesFileScanTask: * spec_id * partition *
+     * delete_file_path
+     *
+     * @param convertConstant callback to convert from an Iceberg typed object to an engine specific
+     *     object
+     * @return a map of column id to constant values returned by this task type
+     */
+    Map<Integer, ?> constantsMap(BiFunction<Type, Object, Object> convertConstant);
+  }
+
+  static class PositionDeleteScanTaskImpl
```

Comment:
I'd say this should become a package private class in `core` and be named `BaseXXX` to be consistent with other tasks.

Reply:
Moved to top level